argo/base/cluster-role.yaml

```diff
@@ -37,12 +38,30 @@ rules:
   - argoproj.io
   resources:
   - workflows
+  - workflows/finalizers
   verbs:
   - get
   - list
   - watch
   - update
   - patch
+  - delete
```

Comment:
It is strange that delete was not included in the earlier manifests. I would want to verify if this has some context from a pipelines team member.

/cc @gaoning777 @Ark-kun @IronPan